### ISSUE TITLE
Feature Request: Add gradient for "BiasAdd" op in C++ API

### ISSUE THREAD

**Issue by dguerra** [tensorflow/tensorflow#12418]
Add BiasAdd gradient implementation for C++ API so it can be added to graph through AddSymbolicGradients function.

**Comment by bpiel** [comment]
@dguerra I was already planning on tackling this. I may have a PR out in the next ~48 hours.

**Comment by dguerra** [comment]
@bpiel Ok, the strange thing is that BiasAddGrad already appears in the C++ API documentation. However it's not added properly to the graph through AddSymbolicGradients.  @suharshs do you know why this might be possible?

**Comment by bpiel** [comment]
@dguerra `BiasAddGrad` already exists as an op, which means the math is already implemented (the hard part). But, `BiasAddGrad` can't be used by `AddSymbolicGradients` without a bit of wrapper logic and then being registered as the gradient of `BiasAdd`.

**Comment by bpiel** [comment]
@dguerra PR submitted. https://github.com/tensorflow/tensorflow/pull/12448

**Comment by dguerra** [comment]
@bpiel Great, I see. So other neural networks gradient ops such as Conv2DBackpropFilter can be added in a similar fashion ?

**Comment by bpiel** [comment]
@dguerra Yes. In fact, I have the code for Conv2D (and MaxPool) in a branch already, but I've only submitted two PRs to TF ever (including this one) and I'd like to wait to get a review on those before submitting more. I don't want to pummel anyone with repeated mistakes.

**Comment by dguerra** [comment]
@bpiel  Thank you!

**Comment by dguerra** [comment]
Hi @bpiel , do you expect to add PR to the other ops you already have in your code?

**Comment by bpiel** [comment]
@dguerra Yeah, I had been waiting for my other PR to go through, but I didn't expect it to take so long. There's been test infrastructure failures.
https://github.com/tensorflow/tensorflow/pull/12391

I'll give up on that and just get the rest out. Should be in the next 24 hours.
FYI, I have:
Conv2D
MaxPool
Mean
Sum

**Comment by dguerra** [comment]
That'd be great :)
